pkg/prometheus/operator.go

```diff
@@ -2121,34 +2120,29 @@ func (c *Operator) createOrUpdateWebConfigSecret(ctx context.Context, p *monitor
 
 	webConfig, err := webconfig.New(
 		webConfigDir,
-		WebConfigSecretName(p.Name),
+		webConfigSecretName(p.Name),
 		tlsConfig,
 	)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "init web config instance failed")
 	}
 
-	ownerReference := metav1.OwnerReference{
+	secretClient := c.kclient.CoreV1().Secrets(p.Namespace)
+	webConfigOwnerReference := metav1.OwnerReference{
 		APIVersion:         p.APIVersion,
 		BlockOwnerDeletion: &boolTrue,
 		Controller:         &boolTrue,
 		Kind:               p.Kind,
 		Name:               p.Name,
 		UID:                p.UID,
 	}
+	webConfigLabels := c.config.Labels.Merge(managedByOperatorLabels)
 
-	secretLabels := c.config.Labels.Merge(managedByOperatorLabels)
-	secret, err := webConfig.MakeConfigFileSecret(secretLabels, ownerReference)
-	if err != nil {
+	if err := webConfig.CreateOrUpdateWebConfigSecret(ctx, secretClient, webConfigLabels, webConfigOwnerReference); err != nil {
 		return err
```

Comment:
if we wrap the error above, we should wrap this one too.

```suggestion
		return errors.Wrap(err, "failed to reconcile web config secret")
```